source/math/FloatExtension.ooc

```diff
@@ -110,7 +110,7 @@ extend Float {
 				value /= 10.0f
 				radix *= 10.0f
 			}
-			while (Float absolute(value) < pow(10.0f, valueDigits-1)) {
+			while (Float absolute(value) - pow(10.0f, valueDigits-1) < -0.000001) {
```

Comment:
It loops while value is smaller than 10^(N-1) and adjusts value and radix so that the value has N digits before the decimal point. Radix represents the factor that, if multiplied with decomposed value, equals the original value.

It was modified because the behaviour wasn't correct in some cases. I don't remember the numbers exactly but an example of such a case would be:

Say that the number 0.100001 is actually represented as 0.09999999 in a Float and that we want 1 value digit. Then we would expect the returned value to be 1.00001. But the case if using value < pow in the while loop is that it will be executed two times and have a value of 9.999999 after the loop.  

If we instead use value - pow < threshold:
After the first iteration of the loop value will be 0.9999999. 
0.9999999 - 1 = -0.0000001 which is not smaller than -1.0E-6 and consequently the result will be 0.9999999. This is much closer to the intended 1.00001 than in the first case.